SFR1.py

```diff
@@ -0,0 +1,228 @@
+        return len(self.sequence)
+
+    def __getitem__(self, index) -> str:
+        return self.sequence[index]
+
+    def __str__(self) -> str:
+        return self.sequence
+
+    def alphabet_is_valid(self) -> bool:
+        dna_alphabet = set("ATGCatgc")
+        rna_alphabet = set("AGCUagcu")
+        if isinstance(self, DNASequence):
+            return set(self.sequence).issubset(dna_alphabet)
+        if isinstance(self, RNASequence):
+            return set(self.sequence).issubset(rna_alphabet)
+        return False
+
+    def complement(self) -> 'NucleicAcidSequence':
+        complement_pairs = {
+            'A': 'T', 'T': 'A', 'G': 'C', 'C': 'G', 'U': 'A',
+            'a': 't', 't': 'a', 'g': 'c', 'c': 'g', 'u': 'a'
+        }
+        complement_sequence = ''
+        for base in self.sequence:
+            complement_sequence += complement_pairs.get(base, base)
+
+        return type(self)(complement_sequence)
+
+    def gc_content(self) -> float:
+        gc_count = 0
+        for base in self.sequence:
+            if base in {'G', 'g', 'C', 'c'}:
+                gc_count += 1
+        total_bases = len(self.sequence)
+        gc_percentage = gc_count / total_bases
+        return gc_percentage
+
+
+class DNASequence(NucleicAcidSequence):
+    def __init__(self, sequence: str):
+        super().__init__(sequence)
+        self.sequence = sequence
+
+    def transcribe(self) -> 'RNASequence':
+        transcribe_sequence = ''
+        for base in self.sequence:
+            if base == 'T':
+                transcribe_sequence += 'U'
+            if base == 't':
+                transcribe_sequence += 'u'
+            else:
+                transcribe_sequence += base
+        return RNASequence(transcribe_sequence)
+
+
+class RNASequence(NucleicAcidSequence):
+    def __init__(self, sequence: str):
+        super().__init__(sequence)
+        self.sequence = sequence
+
+
+class AminoAcidSequence(BiologicalSequence):
+    def __init__(self, sequence: str):
+        self.sequence = sequence
+
+    def __len__(self) -> int:
+        return len(self.sequence)
+
+    def __getitem__(self, index) -> str:
+        return self.sequence[index]
+
+    def __str__(self) -> str:
+        return self.sequence
+
+    def alphabet_is_valid(self) -> bool:
+        amino_acid_alphabet = set("ARNDCHGQEILKMPSYTWFV")
+        return set(self.sequence).issubset(amino_acid_alphabet)
+
+    def calculate_molecular_weight(self) -> float:
+        amino_acid_weights = {
+            'G': 57.051, 'A': 71.078, 'S': 87.077, 'P': 97.115, 'V': 99.131,
+            'T': 101.104, 'C': 103.143, 'I': 113.158, 'L': 113.158, 'N': 114.103,
+            'D': 115.087, 'Q': 128.129, 'K': 128.172, 'E': 129.114, 'M': 131.196,
+            'H': 137.139, 'F': 147.174, 'R': 156.186, 'Y': 163.173, 'W': 186.210
+        }
+        molecular_weight = 0.0
+        for aa in self.sequence:
+            molecular_weight += amino_acid_weights[aa]
```

Comment:
Код верный, но биологически при вычислении массы белка нужно вычитать воду, например:
```suggestion
    list_input_seq = list(seq)
    water_mw = 18
    for aa in list_input_seq:
        total_mw = sum(aa_weight_dict[a] for a in list_input_seq)
        mw_water_removed = (total_mw - (water_mw * (len(list_input_seq)-1)))
    return mw_water_removed
```